docs/en/observability/threshold-alert.asciidoc

```diff
@@ -0,0 +1,152 @@
+* *Define query filter (optional):* Use a query filter to narrow down the data that the rule applies to. For example, set a query filter to a specific host name--`host.name:host-1`--to only apply the rule to that host.
+
+[discrete]
+[[custom-threshold-rule-conditions]]
+== Set rule conditions
+
+Set the conditions for the rule to detect using aggregations, an equation, and a threshold.
+
+[discrete]
+[[custom-threshold-aggregation]]
+=== Set aggregations
+
+Aggregations summarize your data to make it easier to analyze.
+Set any of the following aggregation types to gather data to create your rule.
+
+* *Average:* The average value of a numeric field.
+* *Max:* The highest value of a numeric field.
+* *Min:* The lowest value of a numeric field.
+* *Cardinality:* The approximate number of unique values in a field.
+* *Document count:* The total number of documents in a field.
+* *Sum:* The total of a numeric field in your dataset.
+
+For example, to gather the total number of log documents with a log level of `warn`:
+
+. Set the *Aggregation* to *Document count*, and set the *KQL Filter* to `log.level: "warn"`.
+. Set the threshold to `IS ABOVE 100` to trigger an alert when the number of log documents with a log level of `warn` reaches 100.
+
+The following section contains more information on setting equations and thresholds.
+
+[discrete]
+[[custom-threshold-equation]]
+=== Set the equation and threshold
+
+Set an equation using your aggregations. Based on the results of your equation, set a threshold to define when to trigger an alert. The equations use basic math or boolean logic. Refer to the following examples for possible use cases.
+
+[horizontal]
+*Math equation*:: Add, subtract, multiply, or divide your aggregations to define conditions for alerting.
++
+*Example* +
+Set an equation and threshold to to trigger an alert when a metric is above a threshold. For this example, we'll use average CPU usage--the percentage of CPU time spent in states other than `idle` or `IOWait` normalized by the number of CPU cores--and trigger an alert when CPU usage is above a specific percentage. To do this, set the following aggregations, equation, and threshold:
++
+. Set the following aggregations:
+** *Aggregation A:* Average `system.cpu.user.pct`
+** *Aggregation B:* Average `system.cpu.system.pct`
+** *Aggregation C:* Max `system.cpu.cores`.
+. Set the equation to `(A + B) / C * 100`
+. Set the threshold to `IS ABOVE 95` to alert when CPU usage is above 95%.
+*Boolean logic*:: Use conditional operators and comparison operators with you aggregations to define conditions for alerting.
++
+*Example* +
+Set an equation and threshold to trigger an alert when two fields that should have same value differ. For this example, we'll use `kubernetes.statefulset.replicas.ready` and `kubernetes.statefulset.replicas.desired`, and trigger an alert when their values differ. To do this, set the following aggregations, equation, and threshold:
```

Comment:
@framsouza is this example a good one to be used in the documentation of custom threshold rule, or do you have another example in mind? ([link](https://observability-docs_3314.docs-preview.app.elstc.co/guide/en/observability/master/custom-threshold-alert.html#custom-threshold-equation))

What is a good description of what can be achieved in this use case?

Reply:
@maryam-saeidi one example that i can think of is related to kuberetes statefulset, something like:

This will trigger if a stetefulset pods running is different from the stateful pods desired.

**Aggregation A**: SUM, `kubernetes.statefulset.ready `
**Aggregation B**:  SUM,` kubernetes.statefulset.desired`
**Equation**:  `A == B ? 1 : 0 `
**Threshold**: `IS BELOW 0 `

Here you can group by the name of cluster and statefulset deployment if you wish.

Reply:
Last point, how about changing
> Set an equation and threshold to trigger an alert when two fields that should have same value differ.

to

> Set an equation and threshold to trigger an alert when the number of stateful pods differs from the number of desired ones.

Reply:
Done, thanks for all of your help on this one!